sklearn/cluster/optics_.py

```diff
@@ -504,6 +507,84 @@ def extract_dbscan(self, eps):
         return _extract_dbscan(self.ordering_, self.core_distances_,
                                self.reachability_, eps)
 
+    def extract_sqlnk(self, maxima_ratio=None,
+                      rejection_ratio=None, similarity_threshold=None,
+                      significant_min=None, min_cluster_size=None,
+                      min_maxima_ratio=None):
+        """Performs automatic cluster extraction for variable density data.
+        All parameters will use the value present in the class instance if
+        not provided.
```

Comment:
Maybe add a reference, was this published anywhere?
Is this an implementation of
> Sander, Jörg, Xuejie Qin, Zhiyong Lu, Nan Niu, and Alex Kovarsky. "Automatic extraction of clusters from hierarchical clustering representations." In Advances in Knowledge Discovery and Data Mining, pp. 75-87. Springer Berlin Heidelberg, 2003.
or yet another approach? Where does the name "sqlnk" come from?

Reply:
Yes, it is an implementation [with many changes at this point] provided [here](https://github.com/amyxzhang/OPTICS-Automatic-Clustering), of the paper you mention. I have been planning to check the references in the documentation and the docstrings, probably in a different PR.